### PR TITLE
ChangeLog and testsuite adjustments

### PR DESCRIPTION
**Pull request by ddeclerck**
As requested [here](https://github.com/OCamlPro/gnucobol/pull/147#discussion_r1763582461) and [here](https://github.com/OCamlPro/gnucobol/pull/147#discussion_r1764849089).

Also it seems at some point recently some changes broke the CI - I'll investigate and update this PR.

### CODE REVIEW

**Comment by GitMensch** [comment]
the test looks good, but it seems something is indeed broken in the parser ?!?

**Comment by ddeclerck** [comment]
> the test looks good, but it seems something is indeed broken in the parser ?!?

Yeah but this is unrelated to the conflicts mentionned in the GC3-GC4 merge PR.
It seems `USAGE PROGRAM-POINTER` with a program-prototype argument has never been parsed correctly.

**Comment by GitMensch** [comment]
Are you working on this?

**Comment by ddeclerck** [comment]
> Are you working on this?

I'd rather postpone this issue for now for I fear it would take me too far away from the GC3-GC4 merge (plus I'm already being taken away by PR 109).

**Comment by ddeclerck** [comment]
> Make it an expected failure then and commit upstream, but don't merge it yet - just verify manually that the result with your GC4merge is not worse.

Maybe I can split the 3 cases POINTER/FUNCTION-POINTER/PROGRAM-POINTER so that we can at least "protect" what already works ?

**Comment by GitMensch** [comment]
Good idea. Possibly with a comment to combine again later. Can directly go upstream.

File: tests/testsuite.src/syn_definition.at [hunk 1]
```diff
@@ -2878,3 +2878,76 @@ prog.cob:6: error: 'DEPTHDR' is not defined
...
+       01 data-type PIC X(10) USAGE DISPLAY IS TYPEDEF.
+       01 data-ptr1 USAGE POINTER data-type.
+       01 data-ptr2 USAGE POINTER TO data-type.
+       PROCEDURE DIVISION.
+           STOP RUN.
+       END PROGRAM prog.
+])
+
+AT_CHECK([$COMPILE_ONLY -std=default -Wno-pending prog.cob], [0], [], [])
+
```
Comment (GitMensch) [inline tests/testsuite.src/syn_definition.at:2900]: drop the empty line here as wall as -std=default, only have the -Wno-pending if needed, same for the other tests

File: tests/testsuite.src/syn_definition.at [hunk 2]
```diff
@@ -2878,3 +2878,76 @@ prog.cob:6: error: 'DEPTHDR' is not defined
 
 AT_CLEANUP
 
+
+# Note: the three tests below should be combined in a single one (when fully supported)
```
Comment (GitMensch) [inline tests/testsuite.src/syn_definition.at:2882]: hm. let's drop the comment and rather extend the checks later (using the pointers)

**Review by GitMensch — APPROVED** [review]
good to go for upstream, can also be included in the GC4 merge (the CI fixes and it isn't bad to have the tests as-is in there as well), can be a bit improved, see below, and then checked in

**Pull request closed without merging**